syntax = "proto3";


service WeightsTransfer {
    rpc SendWeights(ClientGrpcMessage) returns (TransferStatus);
}

message ClientGrpcMessage {
  message SendParameters {
    string client_id = 1;
    int64 train_dataset_length = 2;
    map<string, bytes> new_model_weight = 3;
  }
  message GetNewVersion {
    string version_path = 1;
  }
  oneof msg {
    SendParameters send_parameters = 1;
    GetNewVersion get_new_version = 2;
  }
}

message TransferStatus {
    bool code = 1;
    string message = 2;
}